lib/mongodb.js

```diff
@@ -486,7 +486,7 @@ MongoDB.prototype.buildWhere = function (model, where) {
       }
       else {
         query[k] = {};
-        query[k]['$' + spec] = cond;
+        query[k][spec[0] !== '$' ? '$' + spec : spec] = cond;
```

Comment:
I wonder if you should check `allowExtendedOperators ` setting as https://github.com/srguiwiz/loopback-connector-mongodb/blob/workaround-issues-99-and-107/lib/mongodb.js#L321.